src/interceptors/Socket/MockSocket.ts

```diff
@@ -22,6 +22,10 @@ export class MockSocket extends net.Socket {
     super()
     this.connecting = false
     this.connect()
+
+    this._final = (callback) => {
+      callback(null)
```

Comment:
Implement the `_final` method so socket writes transition the socket to the right "final" state:

- set `writableFinished` to `true`
- emit `finish` event

See: https://github.com/nodejs/node/blob/3a456c6db802b5b25594d3a9d235d4989e9f7829/lib/internal/streams/writable.js#L907

> `this._final()` is called on `prefinish`.